sklearn/preprocessing/data.py

```diff
@@ -984,10 +984,7 @@ def transform(self, X, y=None):
 
         if sparse.issparse(X):
             if self.with_scaling:
-                if X.shape[0] == 1:
-                    inplace_row_scale(X, 1.0 / self.scale_)
-                elif self.axis == 0:
-                    inplace_column_scale(X, 1.0 / self.scale_)
+                inplace_column_scale(X, 1.0 / self.scale_)
```

Comment:
It should also raise an error if X is sparse and `self.with_centering` is set to True, no?